server/command_test.go

```diff
@@ -1,1088 +1,916 @@
 package main
 
 import (
+	"encoding/json"
+	"fmt"
 	"testing"
 	"time"
 
-	"github.com/mattermost/mattermost-plugin-msteams/server/metrics"
-	mockMetrics "github.com/mattermost/mattermost-plugin-msteams/server/metrics/mocks"
 	"github.com/mattermost/mattermost-plugin-msteams/server/msteams/clientmodels"
-	mockClient "github.com/mattermost/mattermost-plugin-msteams/server/msteams/mocks"
-	mockStore "github.com/mattermost/mattermost-plugin-msteams/server/store/mocks"
 	"github.com/mattermost/mattermost-plugin-msteams/server/store/storemodels"
-	"github.com/mattermost/mattermost-plugin-msteams/server/testutils"
 	"github.com/mattermost/mattermost/server/public/model"
-	"github.com/mattermost/mattermost/server/public/plugin/plugintest"
 	"github.com/pkg/errors"
 	"golang.org/x/oauth2"
 
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/mock"
+	"github.com/stretchr/testify/require"
 )
 
+func assertNoCommandResponse(t *testing.T, actual *model.CommandResponse) {
+	require.NotNil(t, actual)
+	require.Equal(t, &model.CommandResponse{}, actual)
+}
+
+func assertCommandResponse(t *testing.T, text string, actual *model.CommandResponse) {
+	require.NotNil(t, actual)
+	require.Equal(t, &model.CommandResponse{
+		ResponseType: model.CommandResponseTypeEphemeral,
+		Text:         text,
+	}, actual)
+}
+
+func assertEphemeralMessage(t *testing.T, websocketClient *model.WebSocketClient, channelID, message string) {
```

Comment:
There's opportunities to share this code in the `public` submodule and make these patterns more reusable. I dream of a day plugins can import all the helper functions they need and focus on just the business semantics of the test.